rush-plugins/rush-operation-resource-plugin/README.md

```diff
@@ -0,0 +1,102 @@
+  "resourcePools": [
+    {
+      "poolName": "android-devices",
+      "resources": [
+        "YOGAA1BBB412",
+        "DROID1ABBA44"
+      ]
+    }
+  ]
+}
+```
+
+Configured this way, _most_ build and test phases will run normally, but only _2_ test operations on projects tagged `android` can run at the same time. When the test scripts for these projects are launched, the environment variable `ANDROID_ID` will be set to the chosen resource.
+
+### Use Case 2: Expensive Builds
+
+In this use case, we've configured our CI/CD to run on a 32-core machine, and want to make maximum usage of it, but there are few troublesome projects that use so much RAM that if they happen to execute in parallel, they can cause intermittent issues. To work around this problem, we can assign a special tag to these projects, and allow only 1 of them to build at once.
+
+```json
+{
+  "resourceConstraints": [
+    {
+      "appliesTo": {
+        "phaseName": "_phase:build",
+        "tagName": "expensive-build"
+      },
+      "resourcePool": {
+        "poolName": "expensive-builds"
+      }
+    }
+  ],
+  "resourcePools": [
+    {
+      "poolName": "expensive-builds",
+      "resourceCount": 1
+    }
+  ]
+}
+```
+
+Note that behind the scenes, specifying `resourceCount` instead of `resources` will simply automatically generate a list of resources (`expensive-builds-1`, etc.). In this case we don't care about exact resources, just the number of parallel builds, so we've left off the optional `envVarName` property.
+
+### Use Case 3: Distinguishing between Local and CI
+
+The plugin configuration file is a simple JSON file, and doesn't offer any run-time configuration options. To simplify the experience for local developers, ensure that your checked-in config file makes sense when building and testing locally, and then overwrite the file in your CI/CD pipeline.
+
+For example, you might add a line like this before running `rush` in CI/CD:
+
+```bash
+cp common/ci/rush-operation-resource-plugin.ci.json common/config/rush/rush-operation-resource-plugin.json
+```
+
+## Implementation Details
+
+Note that this plugin does not attempt to change the way Rush calculates the build order of a given set of operations. (Rush uses the dependency tree of your monorepo's project set, and the defined phase dependencies of any phases included in the current command, to construct a relatively optimal build graph, and this plugin does not change it.)
+
+Instead, this plugin makes use of Rush's built-in phased operation hooks to _delay_ the start of a given task if it meets certain criteria. This approach has pros and cons:
+
+ - One con is that this approach is not always _optimal_. A given operation may need to be delayed because it requires a resource where none is available, and it will simply wait for another operation to finish so it can start. An optimal implementation would pause this operation and run a _different_ operation, perhaps one with no resource constraints.
```

Comment:
You could leverage `beforeExecuteOperation` with early return to let the scheduler schedule something else.